interpreter/code_interpreters/dockerfiles/dev_note.md

```diff
@@ -0,0 +1,22 @@
+=======================================================================
+                              NOTE TO DEVELOPERS
+=======================================================================
+
+Please avoid manually editing the following files:
+- Dockerfile
+- requirements.txt
+- hash.json
+
+These files are key components for Open-Interpreter's containerized execution features. Manually editing them can disrupt the program's ability to:
+
+1. Know when to rebuild the Docker image.
+2. Perform other related functionalities efficiently.
+
+If you need to make adjustments, kindly use the 'DockerManager' class. It offers convenient methods like:
```

Comment:
This would benefit from some examples and more thorough information as it's unclear how to work with this and when a developer would want/need to.

It might also be a good idea to a drop a link to this document and some notes about Containerization in [`CONTRIBUTING.md`](https://github.com/KillianLucas/open-interpreter/blob/main/CONTRIBUTING.md) once the container functionality is ready to merge.

Reply:
I'm going to resolve this one and we can focus on it as a follow up once we get this MVP merged in and don't have to fight against merge conflicts with such a large changeset.